src/python/pants/backend/terraform/tool.py

```diff
@@ -450,6 +452,17 @@ async def setup_terraform_process(
     def prepend_paths(paths: Tuple[str, ...]) -> Tuple[str, ...]:
         return tuple((Path(request.chdir) / path).as_posix() for path in paths)
 
+    # Initialise the Terraform provider cache, since Terraform expects the directory to already exist.
+    await Get(
```

Comment:
Separate invocations of `mkdir` and `terraform` would not work in remote execution because the `mkdir` and `terraform` invocations could potentially end up on separate workers. Moreover, it can even be in an issue in local execution because the `mkdir` could be cached and the `terraform` invocation uncached. Thus, the side effect of making the cache directory would not occur because Pants would see the `mkdir` process as cached.

They should be a singe process invocation running the shell with a custom script (doing the necessary replacements):

```
script = f"""mkdir -p {tf_plugin_cache_dir} && exec  __terraform/terraform  -chdir={shlex.quote(request.chdir)} {shlex.join(request.args)}"""

process = Process(argv=(bash.path, "-c", script), ...)
```

Reply:
Note: `bash.path` comes from a `bash: BashBinary` instance.

Reply:
oh, that makes a lot of sense. I've fixed it, thank you for the direction on how.